spec/classes/init_spec.rb

```diff
@@ -351,97 +348,19 @@
-      end
-    end
-
-    context 'when major release of Amazon Linux is unsupported' do
-      let :facts do
-        { osfamily:                  'RedHat',
-          operatingsystem:           'Amazon',
-          operatingsystemmajrelease: '3',
-          monit_version:             '5' }
-      end
-
-      it 'fails' do
-        expect {
-          is_expected.to contain_class('monit')
-        }.to raise_error(Puppet::Error, %r{monit supports Amazon Linux 2\. Detected operatingsystemmajrelease is <3>})
-      end
-    end
-
-    context 'when major release of EL is unsupported' do
-      let :facts do
-        { osfamily:                  'RedHat',
-          operatingsystem:           'CentOS',
-          operatingsystemmajrelease: '4',
-          monit_version:             '5' }
-      end
-
-      it 'fails' do
-        expect {
-          is_expected.to contain_class('monit')
-        }.to raise_error(Puppet::Error, %r{monit supports EL 5, 6 and 7\. Detected operatingsystemmajrelease is <4>})
-      end
-    end
-
-    context 'when major release of Debian is unsupported' do
-      let :facts do
-        { osfamily:                  'Debian',
-          operatingsystemmajrelease: '4',
-          lsbdistcodename:           'etch',
-          monit_version:             '5' }
-      end
-
-      it 'fails' do
-        expect {
-          is_expected.to contain_class('monit')
-        }.to raise_error(Puppet::Error, %r{monit supports Debian 6 \(squeeze\), 7 \(wheezy\), 8 \(jessie\), 9 \(stretch\) and 10 \(buster\) \
-and Ubuntu 10\.04 \(lucid\), 12\.04 \(precise\), 14\.04 \(trusty\), 16\.04 \(xenial\) and 18\.04 \(bionic\)\. \
-Detected lsbdistcodename is <etch>\.})
-      end
-    end
-
-    context 'when major release of Ubuntu is unsupported' do
-      let :facts do
-        { osfamily:                  'Debian',
-          operatingsystemmajrelease: '8',
-          lsbdistcodename:           'hardy',
-          monit_version:             '5' }
-      end
-
-      it 'fails' do
-        expect {
-          is_expected.to contain_class('monit')
-        }.to raise_error(Puppet::Error, %r{monit supports Debian 6 \(squeeze\), 7 \(wheezy\), 8 \(jessie\), 9 \(stretch\) and 10 \(buster\) \
-and Ubuntu 10\.04 \(lucid\), 12\.04 \(precise\), 14\.04 \(trusty\), 16\.04 \(xenial\) and 18\.04 \(bionic\). \
-Detected lsbdistcodename is <hardy>\.})
-      end
-    end
-
-    context 'when osfamily is unsupported' do
-      let :facts do
-        { osfamily:                  'Unsupported',
-          operatingsystemmajrelease: '9',
-          monit_version:             '5' }
-      end
-
-      it 'fails' do
-        expect {
+        expect do
           is_expected.to contain_class('monit')
-        }.to raise_error(Puppet::Error, %r{monit supports osfamilies Debian and RedHat\. Detected osfamily is <Unsupported>\.})
+        end.to raise_error(Puppet::PreformattedError, %r{expects a value of type Undef or Integer\[1})
```

Comment:
This is a repeated pattern. Above I already suggested to rewrite it to `is_expected.to compile.and_raise_error` and I think it's true for the whole test suite.

Reply:
ok, will have a look